---
title: "LAB 2"
author: "Shelby Phillips"
date: "2/4/2020"
output: html_document
---


Vectors Practice
```{r}
SNPs <- c ("AA", "AA", "GG", "AG", "AG", "AA", "AG", "AA", "AA", "AA", "AG")
SNPs
```
Factors Practice
```{r}
SNPs_cat <- factor(SNPs)
SNPs_cat
table(SNPs_cat)
SNPs_cat
plot(SNPs_cat)
as.numeric(SNPs_cat)
```
Matrices Practice
```{r}
Day1 <- c(2,4,6,8)
Day2 <- c(3,6,9,12)
Day3 <- c(1,4,9,16)
A <- cbind(Day1,Day2,Day3)
A
```
```{r}
Day1 <- c(2,4,6,8)
Day2 <- c(3,6,9,12)
Day3 <- c(1,4,9,16)
b <- rbind(Day1,Day2,Day3)
b 
```
```{r}
Day4 <- c(5,10,11,20)
C <- rbind(b,Day4)
C
```
```{r}
A*10
```
```{r}
A[1]
```
```{r}
A[12]
```
```{r}
A[ ,c(1,3)]
```
```{r}
A[c(2,4), ]
```
```{r}
t(A)
```
Data Frames Practice
```{r}
Gene1 <- c(2,4,6,8)
Gene2 <- c(3,6,9,12)
Gene3 <- c(1,4,9,16)
Gene <- c("Day 1", "Day 2","Day 3", "Day 4")
RNAseq <- data.frame(Gene1, Gene2, Gene3, row.names = Gene)
RNAseq
```

```{r}
RNAseq$Gene3
```

```{r}
plot(RNAseq$Gene1,RNAseq$Gene3)
```

```{r}
RNAseq$Gene4 <-c(5,10,15,20)
RNAseq
```
```{r}
RNAseq[,"gene5"] <-c(1,2,3,3)
RNAseq
```
```{r}
RNAseq["Day 4",] <- rbind(10,14,20,22,3)
```
Checking on object types practice
```{r}
x=1
str(x)
```
```{r}
a="ATGCCCTGA"
str(a)
```
```{r}
str(SNPs)
```
```{r}
SNPs <- c("AA", "AA", "GG", "AG", "AG", "AA","AG", "AA", "AA", "AA", "AG")
str(SNPs_cat)
```
```{r}
Day1 <- c(2,4,6,8)
Day2 <-c(3,6,9,12)
Day3 <-c(1,4,9,16)
B <- rbind(Day1,Day2,Day3)
B<- rbind(Day1,Day2,Day3)
str(B)
```
```{r}
Gene1 <-c(2,4,6,8)
Gene2 <-c(3,6,9,12)
Gene3 <-c(1,4,9,16)
Gene <-c("Day1","Day2","Day3","Day4")
RNAseq <- data.frame(Gene1,Gene2,Gene3,row.names = Gene)
str(RNAseq)
```
Getting information on a dataset practice
```{r}
SNP_table <- read.table("23andMe_example_cat25.txt", header = TRUE, sep = "\t")
SNP_table
```

```{r}
names(SNP_table)
```
```{r}
str(SNP_table)
```
```{r}
levels(SNP_table$genotype)
```
```{r}
dim(SNP_table)
```
```{r}
class(SNP_table)
```
```{r}
SNP_table
```
```{r}
head(SNP_table, n=10)
```
```{r}
tail(SNP_table, n=5)
```
```{r}
SNP_table$chromosome <- as.factor(SNP_table$chromosome)
str(SNP_table)
```
```{r}
SNP_table$chromosome <- as.integer(SNP_table$chromosome)
str(SNP_table)
```
```{r}
SNP_table_AG <- subset(SNP_table, genotype == 'AG')
SNP_table_AG
```
```{r}
table(SNP_table_AG$chromosome)
```
```{r}
subset(SNP_table,position > 700000 & position < 800000)
```
Exercise 1
```{r}
vector1 <-c(1,3,6,9,12)
vector2 <-c(1,0,1,0,1)
vector1+vector2
vector1-vector2
vector1*vector2
vector1/vector2
```
Exercise 2
```{r}
x <-c(0,1,2,3)
y <-c("aa","bb","cc","dd")
z <-c("aa",1,"bb",2)
str(x)
str(y)
str(z)
```
Exercise 3
```{r}
genotype1 <-c("AA","AA","AG","GG","GG")
genotype2 <-c("AA","AA","GG","GG","GG")
A <-cbind(genotype1,genotype2)
A
genotype1 <-c("AA","AA","AG","GG","GG")
genotype1
genotype1_cat <-factor(genotype1)
genotype1_cat
table(genotype1_cat)
genotype2 <-c("AA","AA","GG","GG","GG")
genotype2
genotype2_cat <-factor(genotype2)
genotype2_cat
table(genotype2_cat)
```
Exercise 4
```{r}
treatment1 <-c(0,1,2,3,4)
treatment2 <-c(0,2,4,6,8)
treatment3 <-c(0,3,6,9,12)
time <-c(0,2,4,6,8)
treatment <-data.frame(treatment1,treatment2,treatment3, row.names = time)
treatment
treatment$time <-c(0, 2, 4, 6, 8)
treatment
plot(treatment$time,treatment$treatment3)
```
Exercise 5
```{r}
SNP_table <-read.table("23andMe_complete.txt", header = TRUE, sep= "\t")
SNP_table
names(SNP_table)
str(SNP_table)
levels(SNP_table$genotype)
dim(SNP_table)
SNP_table$chromosome <- as.factor(SNP_table$chromosome)
str(SNP_table)
```
there are 22 chromosomes, plus the X and Y chromosomes, and MT, so there are 25 levels for the chromosome in the complete file. This differs from the example because the example only included SNPs from one chromosome. Whearas the complete file includes SNPs from 22 chromosomes, the sex chromosomes, and the MT group.

Exercise 6
```{r}
SNP_table$chromosome <- as.integer(SNP_table$chromosome)
str(SNP_table) 
levels(SNP_table$genotype)
SNPs <-c("--", "A", "AA", "AC", "AG", "AT", "C", "CC", "CG", "CT", "D", "DD", "DI", "G", "GG", "GT", "I", "II", "T", "TT")
SNPs
SNPs_cat <-factor(SNPs)
SNPs_cat
table(SNPs_cat)
str(SNPs_cat)
SNP_table <-read.table("23andMe_complete.txt", header = TRUE, sep= "\t")
SNP_table
str(SNP_table)
SNPs_complete <-factor(SNP_table$genotype)
SNPs_complete
table(SNPs_complete)
```
I AM SOOOOOO PUMPED, THAT TOOK ME AGES TO FINALLY MAKE HAPPEN :)


Exercise 7
```{r}
SNP_table_A <- subset(SNP_table, genotype == 'A') 
SNP_table_A
table(SNP_table_A$chromosome)
```
the single letter genotype A is found on the X and Y chromosomes and MT chromosome


